src/types/opportunitiesTypes.ts

```diff
@@ -0,0 +1,10 @@
+export interface OpportunityMinimal {
+  title: string;
+}
+
+export interface Opportunity extends OpportunityMinimal {
```

Comment:
In TypeScript the way to signal this is to define `Opportunity` and specify optional fields where appropriate using `?`.

e.g.

```
export interface Opportunity {
  id?: number;
  title: string;
  createdAt?: Date;
}
```

For situations where we only need a smaller piece of an opportunity there are a series of [utility types](https://www.typescriptlang.org/docs/handbook/utility-types.html) that let us specify a *contextual* modifications of a given interface.

For instance if we had an API endpoint that ONLY needed a smaller subset of `Opportunity` fields we could use `Pick<Opportunity, 'title' | 'id'>` and that would require both title and id for that particular parameter.

Reply:
Thanks for the explanation here and in https://github.com/PhilanthropyDataCommons/service/pull/55#discussion_r927940853. It looks like I clicked the link you gave here but did not actually go through it and understand it until your other comment. I see the approach and it makes sense!